x/stake/keeper.go

```diff
@@ -51,8 +64,12 @@ func (k Keeper) getValidator(store sdk.KVStore, addr sdk.Address) (validator Val
 // set the main record holding validator details
 func (k Keeper) setValidator(ctx sdk.Context, validator Validator) {
 	store := ctx.KVStore(k.storeKey)
+	// set main store
 	bz := k.cdc.MustMarshalBinary(validator)
 	store.Set(GetValidatorKey(validator.Owner), bz)
+	// set pointer by pubkey
+	bz = k.cdc.MustMarshalBinary(validator.Owner)
+	store.Set(GetValidatorByPubKeyKey(validator.PubKey), bz)
```

Comment:
Should stay consistent with other indexes of the validator set and just use the validator owner address bytes directly (no need to marshal already bytes)
```
store.Set(GetValidatorByPubKeyKey(validator.PubKey), validator.Owner)
```

Reply:
Also I'm not sure that we want to have this as part of `setValidator` - less optimizable - only need to set this store if the pubKey changes (which it usually doesn't) - I'd rather separate updating the index to a new keeper function `setValidatorByPubkeyIndex(...)`

Reply:
No longer marshals bytes. Agreed on separate function.

Reply:
Separated into `setValidatorByPubKeyIndex`, called on declare-candidacy and genesis initialization.